### DISCUSSION TITLE
Please ensure this will stay a buildless framework.

### DISCUSSION THREAD

**Discussion by pfeatherstone** [alpinejs/alpine#1516] [4. General]
Please, please, please, please, please, please don't ever make apline require a build system.
I started using Vue, only to find out that the recommended way of using it was using .vue files, which require a build step.
Angularjs was great because you could simply add it to your HTML and off you went. Unfortunately, they deprecated it.
Please don't make that same mistake.
For example, if you were to ever support custom directives, please don't define a .alpine file for those custom components. Just use something like ```Alpine.directive(...)```.
It would be awesome if this became the framework of choice for browsers, and for people who don't want to build all their javascript, HTML and CSS into a bundle. Yes, building some typescript will probably give you better performance, but if you're building a few bootstrap web pages, you honestly can't be bothered to build those pages using some fancy IDE. You just want a text editor.
So far, it looks like alpine fits the bill quite nicely. It would be even nicer if this was a future guarantee.

**Comment by sqllyw** [comment]
Alpine.directive remind me of good old days of Angularjs 1, I did try Vue after Angular 2, now I stay in Alpinejs, alpine is enough for my needs.

**Comment by pfeatherstone** [comment]
If people don't agree, it would be great to read their opinion. And if you DO want to see a build system and some ```.alpine``` files, can you describe why you don't just use Angular, Vue, React or some other framework I'm not aware of, which do precisely that.

**Comment by KevinBatdorf** [comment]
I doubt Caleb would change this since he states in every video that it's one of the core tenants of Alpine.

But you will need to make custom builds if you want a specific build of Alpine (maybe without some directives).

I don't think you need a build process for Vue either, including .vue files. I remember there being a loader for those, or you could just inline your Vue code like Alpine, I believe. Don't quote me there though.

**Reply by pfeatherstone** [reply]
Yeah, you can definitely use Vue without ```.vue``` files, but every single Vue library uses them. Like every single useful wrapper uses a ```.vue``` file. As far as a browser-side loader, i've never seen that before.

**Reply by pfeatherstone** [reply]
And yeah, if you want to build the alpine library itself, then yeah, that's fine. But having to build your code in order to use alpine would be naf.

**Comment by jimmywarting** [comment]
Buildless tool, no jsx, or typescript was the main reason why i steered towards alpine.js. 
it's possible to achieve same level of type support without TS. (using jsdoc and other tech).

Also wish that alpine js stays "buildless"

**Comment by ekwoka** [comment]
Honestly, I feel this is such a nonsense concern.

Are you just writing a bunch of purely raw html files?

It's likely you are already using some kind of build process. and those processes take only like 1 second to set up and have functional anyway.

But i don't think there is any risk that alpinejs will ever not be valid js, doesn't make sense with anything of how it works.

So even if lots of benefits come from ESM, you can use those in the browser anyway.

@jimmywarting I'm not sure what "no typescript" benefit is. Since typescript projects are basically always distributed in js anyway.

**Reply by jimmywarting** [reply]
unfortunately not all TS projects are distributed in purely distributed in JS. 
- some build project solo for Deno
- Some don't compile to ESM compatible code and use extensionless import
- some bundle all there code into one single file, those making it harder to import just the things you need

TypeScript will never be supported by browsers and likely never will. it's just another flow / coffeescript (compile to js first before it can even run). If a typed language is so desired. then i think it's better to stick to c/c++ or any other typed language and then compile it to WASM
You can't simply copy-paste some piece of code and have it run anywhere. 

And yes i do build purely raw html files sometimes, then i do not use build process at all. and the do not take just 1s to build.

**Reply by ekwoka** [reply]
the third one there is actually an issue with Alpine lol it's all pre-bundled.

And all of those are fine with just importing from an esm cdn if you want no build steps (stuff for Deno as a runtime won't work in the browser anyway, just like node stuff)

> If a typed language is so desired. then i think it's better to stick to c/c++ or any other typed language and then compile it to WASM

Huh what? That's quite literally nonsense. WASM has sizable overhead, and even better languages like Rust have issues being particularly good in WASM.

> and the do not take just 1s to build.

With vite, it's even faster :D

Alpine takes like 100ms to build with esbuild.

There's basically no reason to have any real project that doesn't use some kind of build step (even if just templating) so it's just a very irrational concern, especially when CDNs exist that will bundle/transpile the ESM stuff anyway.

but yeah, I don't think Alpine will ever even consider adding anything that requires such transpilation by the consuming project, even if TS is the future.
